lib/firmware.ex

```diff
@@ -84,12 +84,15 @@ defmodule Nerves.Firmware do
   also sets the firwmare state based on the action to reflect the update, and
   prevent multiple updates from overwriting known good firmware.
 
+  * `action` can be one of `:upgrade` or `:complete`
+  * `args` is a list of extra arguments to be passed to fwup.
+
   Returns {:error, :await_restart} if the upgrade is requested after
   already updating an image without a reboot in-between.
   """
-  @spec apply(String.t, atom) :: :ok | {:error, reason}
-  def apply(firmware, action) do
-    GenServer.call @server, {:apply, firmware, action}
+  @spec apply(String.t, atom, args \\ []) :: :ok | {:error, reason}
+  def apply(firmware, action, args) do
```

Comment:
Oh interesting. I have this in my local branch. Looks like i just forgot to push.